src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java

```diff
@@ -430,6 +430,12 @@ public abstract class AbstractAssemblyMojo extends AbstractMojo implements Assem
     @Parameter
     private String overrideGroupName;
 
+    /**
+     * Override of umask.
```

Comment:
If this shows up in user docs, better description is needed. If it doesn't no description is needed.

Reply:
I followed description of `overrideGroupName` option here. What about:

```suggestion
     * Override of umask applied to permissions of files/directories before they are put into assembly.
```

?

Reply:
Now that I see the description, does it even matter that this overrides another value? It would be simpler to just say this sets the umask directly, and also specify the default value used if one is not specified here. The rest is implementation detail.

Reply:
Changed to

```java
/**
 * Mask which is applied to permissions of files/directories before they are put into assembly.
 * If {@code null} then the mask is not explicitly configured and remains implementation-specific.
 */
```

When this option is not configured (`null`), then the umask remains Plexus Archiver implementation detail (which I know is `022`, but I cannot put it into documentation of `overrideUmask` configuration option of Maven Assembly Plugin).

IMHO, we can not put `022` as default value of `overrideUmask` configuration option of Maven Assembly Plugin, because it breaks backward compatibility (in case someone uses another version of Plexus Archiver - the one which has different default value of umask option - with Maven Assembly Plugin and migrates to the new version of Maven Assembly Plugin with this pull request integrated).